src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -943,41 +939,40 @@ def _get_default_address_pool(cli_ctx, resource_group, balancer_name, balancer_t
     return values[0]
 
 
-def _validate_vmss_create_load_balancer_or_app_gateway(cmd, namespace):
-    from msrestazure.azure_exceptions import CloudError
-    from msrestazure.tools import parse_resource_id
-    INSTANCE_THRESHOLD = _get_vmss_create_instance_threshold()
+def _validate_vmss_single_placement_group(namespace):
+    if namespace.platform_fault_domain_count is not None and namespace.zones is None:
+        raise CLIError('usage error: --platform-fault-domain-count <COUNT> --zones <ZONES>')
+    if namespace.platform_fault_domain_count == 1 or namespace.zones or namespace.instance_count > 100:
+        if namespace.single_placement_group is None:
+            namespace.single_placement_group = 'false'
+        elif namespace.single_placement_group == 'true':
+            raise CLIError("usage error: '--single-placement-group' should be turned off when one of the conditions"
+                           " are met: 1. Zonal, 2. 100+ instances, 3. fault domain count set to 1")
 
     # convert the single_placement_group to boolean for simpler logic beyond
-    if namespace.single_placement_group is None:
-        namespace.single_placement_group = namespace.instance_count <= INSTANCE_THRESHOLD
-    else:
-        namespace.single_placement_group = (namespace.single_placement_group == 'true')
+    namespace.single_placement_group = (namespace.single_placement_group != 'false')
```

Comment:
You should be able to register the argument to get actual Boolean values instead of strings 'true' and 'false'.  That may simplify your code (or not!).